src/datasets/arrow_dataset.py

```diff
@@ -1384,6 +1384,31 @@ def with_transform(
         dataset.set_transform(transform=transform, columns=columns, output_all_columns=output_all_columns)
         return dataset
 
+    def prepare_for_task(self, task: str) -> "Dataset":
+        """Prepare a dataset for the given task.
+
+        Casts :attr:`datasets.DatasetInfo.features` according to a task-specific schema.
```

Comment:
Let's have a list of the supported tasks here ?

Reply:
good idea! i've added the list of tasks under the `task` arg since it felt a bit more natural to be there